src/presentation/react/core/components/sync-wizard/common/MetadataSelectionStep.tsx

```diff
@@ -122,10 +124,46 @@ export default function MetadataSelectionStep({ syncRule, onChange }: SyncWizard
         });
     }, [compositionRoot, snackbar, syncRule.originInstance]);
 
+    useEffect(() => {
+        if (!_.isEmpty(metadataSyncAll)) {
+            compositionRoot.metadata.getByIds(syncRule.metadataIds, remoteInstance, "id").then(metadata => {
+                const idsFromSyncAllMetadataTypes = _.compact(
+                    _(metadata)
+                        .pick(metadataSyncAll)
+                        .values()
+                        .value()
+                        .flat()
+                        .map(entity => entity?.id)
+                );
+
+                setIdsToIgnore(idsFromSyncAllMetadataTypes);
+            });
+        } else {
+            setIdsToIgnore([]);
+        }
+    }, [compositionRoot.metadata, metadataSyncAll, remoteInstance, syncRule.metadataIds]);
+
     const notifyNewModel = useCallback(model => {
         setModel(() => model);
     }, []);
 
+    const notifyModelSyncAllChange = useCallback(
+        (value: boolean) => {
+            setMetadataSyncAll(types => {
+                const modelName = model.getCollectionName();
+                const syncAllTypes = value ? _.uniq(types.concat(modelName)) : _.without(types, modelName);
+                const ruleTypes = _.uniq(syncRule.metadataTypes.concat(syncAllTypes));
+
+                onChange(syncRule.updateMetadataTypes(ruleTypes).updateMetadataSyncAll(syncAllTypes));
+
+                return syncAllTypes;
+            });
+        },
+        [model, syncRule, onChange]
+    );
+
+    const modelIsSyncAll = useMemo(() => metadataSyncAll.includes(model.getCollectionName()), [metadataSyncAll, model]);
+
```

Comment:
[no need to change] Even when arriving at components with all useEffect/useCallback in-line, it's a good practice to create the new code as hooks, so we go in the right direction. Specially important for inline useEffect, which are nameless blocks of actions.